.github/workflows/release.yml

```diff
@@ -119,11 +117,88 @@ jobs:
       - name: "Push changes from release branch"
         run: git push
 
-      # Pushing the release tag starts build processes that then produce artifacts for the release
       - name: "Push release tag"
         run: git push origin ${{ inputs.version }}
 
       - name: "Set summary"
         run: |
           echo '🚀 Created tag and drafted release for version [${{ inputs.version }}](${{ env.RELEASE_URL }})' >> $GITHUB_STEP_SUMMARY
           echo '✍️ You may now update the release notes and publish the release when ready' >> $GITHUB_STEP_SUMMARY
+
+  static-analysis:
+    needs: prepare-release
+    name: "Run Static Analysis"
+    uses: ./.github/workflows/static-analysis.yml
```

Comment:
By default, I assume this would invoke all jobs in the referenced file and that's why you added a conditional on the Rector job.

I noted that this is re-using an entire workflow ([`jobs.<job_id>.uses`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iduses)), as opposed to an action being imported into a step ([`jobs.<job_id>.steps[*].uses`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsuses)).

I'm a bit confused about how this single `static-analysis` job would map to multiple jobs in `static-analysis.yml` if you hadn't specified a condition to exclude Rector. Would they just be run in sequence? [Reusing Workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) doesn't address this, and the example therein only has a single job defined in the import file. To that end, I wonder if it'd make more sense to split the Psalm and Rector jobs into separate files (e.g. `static-analysis-psalm.yml` and `static-analysis-rector.yml`) if there isn't much commonality between them anyway.

Reply:
Yes - unlike actions that contain a set of functionality to be included in a job, this allows to reuse an entire workflow as a job in a different workflow, without sharing data with any other job.

As for what this looks like, I wasn't sure at first either but it actually looks rather nice: https://github.com/alcaeus/mongo-php-library/actions/runs/9447195881. I'll split the workflow up into separate workflows for psalm and rector.